icu4c/source/test/testdata/localeCanonicalization.txt

```diff
@@ -0,0 +1,1648 @@
+# File copied from cldr common/testData/localeIdentifiers/localeCanonicalization.txt
```

Comment:
yes, I am not sure what you like me to do in this PR for that. 
I believe common/testData/localeIdentifiers/localeCanonicalization.txt is generated by mark but I believe there are no code hook up to automatically generate that inside CLDR when anyone change that file.

Reply:
As I said, I think this should be documented for now. Not sure if we have some place where people actually look, but at a quick glance cldr-icu-readme.txt seemed to make sense because that's where people look who import data from CLDR.

And for the future (via a ticket), it would be better to automate this copying, which would probably be an addition to the cldr-to-icu converter or a sibling tool. (Like Hugo was looking into for unit formatting test data.)